modules/library/main/src/main/java/org/geotools/feature/simple/SimpleFeatureTypeBuilder.java

```diff
@@ -1028,7 +1028,10 @@ public static SimpleFeatureType retype(SimpleFeatureType original, List<String>
 
         // add attributes in order
         for (int i = 0; i < types.size(); i++) {
-            b.add(original.getDescriptor(types.get(i)));
+            AttributeDescriptor descriptor = original.getDescriptor(types.get(i));
+            if (descriptor != null) {
+                b.add(descriptor);
+            }
```

Comment:
Given that I only started getting nullPointerExceptions after I changed PropertyNameResolvingVisitor, I am fairly certain the null is being created from the empty string, I just don't know exactly where.